.azure-pipelines/client.yml

```diff
@@ -58,57 +58,33 @@ jobs:
       displayName: 'Component Detection'
 
     - task: Maven@3
-      displayName: 'Generate project site'
+      displayName: 'Generate Maven project site, including JavaDocs and CheckStyle reports'
       inputs:
         mavenPomFile: $(pomFile)
-        options: '--batch-mode'
+        options: '--batch-mode -DskipTests'
         mavenOptions: '-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
         javaHomeOption: 'JDKVersion'
         jdkVersionOption: '1.11'
         jdkArchitectureOption: 'x64'
         publishJUnitResults: false
-        goals: 'site'
+        goals: 'site:site site:stage'
```

Comment:
Since the checkstyle reporting is now integrated into the site generation goal, and checkstyle reporting requires the com.azure:sdk-build-tools:jar:1.0.0 artifact to be installed into the local repository, this fails on a clean system (no m2 cache) with the following error:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.7.1:site (default-cli) on project azure-client-sdk-parent: failed to get report for org.apache.maven.plugins:maven-checkstyle-plugin: Plugin org.apache.maven.plugins:maven-checkstyle-plugin:3.0.0 or one of its dependencies could not be resolved: Could not find artifact com.azure:sdk-build-tools:jar:1.0.0 in ossrh (https://oss.sonatype.org/content/repositories/snapshots/) -> [Help 1]
```

We should be able to fix it by swapping the order of these two steps - run the install step first, followed by the site generation step. However I'm not clear 1) if that's okay (or if the site really needs to be generated first) and 2) why checkstyle reporting needs the sdk-build-tools jar in the first place.

Reply:
I'll make this change, it should be fine. I'm out of town today, so will change this next week. Thanks.